### PR TITLE
Add services and apis docs for TS

### PR DESCRIPTION
**Pull request by simon-johansson**
First pass at converting this page https://encore.dev/docs/primitives/services-and-apis  into TS docs.

### CODE REVIEW

File: docs/primitives/services-and-apis-ts.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+// getBlogPost retrieves a blog post by id.
+const getBlogPost = api(
+  { method: "GET", path: "/blog/:id" },
+  async ({ id }: { id: number }): Promise<BlogPost> => { 
+	  // Use id to query database...
+  },
+);
+```
+
+You can also combine path parameters with body payloads. For example, if you want to have an `updateBlogPost` endpoint:
```
Comment (simon-johansson) [inline docs/primitives/services-and-apis-ts.md:139]: Is this possible at the moment? To have both a path param and a body payload?

File: docs/primitives/services-and-apis-ts.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+```
+/my-app
+├── encore.app          // ... and other top-level project files
+├── package.json  
+│
+├── hello               // hello service (a folder)
+│   ├── hello.ts        // hello service code
+│   └── hello_test.ts   // tests for hello service
+│
+└── world               // world service (a Go package)
```
Comment (eandre) [inline docs/primitives/services-and-apis-ts.md:25]: Not a go package

File: docs/primitives/services-and-apis-ts.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+and ensure it matches the schema, with zero boilerplate.
+
+To define an API, use the `api` function from the `@encore.dev/api` module to wrap a regular TypeScript async function that receives the request data as input and returns response data.
+This tells Encore that the function is an API endpoint. Encore will then automatically generate the necessary boilerplate at compile-time.
+
+In the example below, we define the API endpoint `ping` which accepts `POST` requests and is exposed as `hello.ping` (because our service name is `hello`).
+
+```typescript
+// inside the hello.ts file
+import { api } from "@encore.dev/api"
```
Comment (eandre) [inline docs/primitives/services-and-apis-ts.md:48]:
```suggestion
import { api } from "encore.dev/api"
```

File: docs/primitives/services-and-apis-ts.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+## Defining APIs
+
+Encore allows you to easily define type-safe, idiomatic TypeScript API endpoints.
+
+It's easy to accept both the URL path parameters, as well as JSON request body data, HTTP headers, and query strings.
+
+It's all done in a way that is fully declarative, enabling Encore to automatically parse and validate the incoming request
+and ensure it matches the schema, with zero boilerplate.
+
+To define an API, use the `api` function from the `@encore.dev/api` module to wrap a regular TypeScript async function that receives the request data as input and returns response data.
```
Comment (eandre) [inline docs/primitives/services-and-apis-ts.md:41]:
```suggestion
To define an API, use the `api` function from the `encore.dev/api` module to wrap a regular TypeScript async function that receives the request data as input and returns response data.
```

File: docs/primitives/services-and-apis-ts.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+In the example below, we define the API endpoint `ping` which accepts `POST` requests and is exposed as `hello.ping` (because our service name is `hello`).
+
+```typescript
+// inside the hello.ts file
+import { api } from "@encore.dev/api"
+
+export const ping = api(
+  { method: "POST" },
+  async ({ name }: PingParams): Promise<PingResponse> => {
```
Comment (eandre) [inline docs/primitives/services-and-apis-ts.md:52]: Maybe a bit too complex syntax for the first example?

```suggestion
  async (p: PingParams): Promise<PingResponse> => {
```

And then use `p.name`?

File: docs/primitives/services-and-apis-ts.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+);
+```
+
+### Request and response schemas
+
+In the example above we defined an API that uses request and response schemas. The request data is of type `PingParams` and the response data of type `PingResponse`. That means we need to define them like so:
+
+```typescript
+// inside the hello.ts file
+import { api } from "@encore.dev/api"
```
Comment (eandre) [inline docs/primitives/services-and-apis-ts.md:64]:
```suggestion
import { api } from "encore.dev/api"
```

File: docs/primitives/services-and-apis-ts.md (hunk 1 above)
Comment (eandre) [reply docs/primitives/services-and-apis-ts.md:139]: Yes

**Review by eandre — APPROVED** [review]

**Pull request merged**